tests/unit/services/test_provider.py

```diff
@@ -241,6 +241,29 @@ def test_get_lxd_provider(monkeypatch, provider_service, lxd_remote, check):
         )
 
 
+@pytest.mark.parametrize(
+    ("platform", "platform_str"),
+    [
+        pytest.param("test-platform", "test-platform", id="simple"),
+        pytest.param(
+            "ubuntu@24.04:amd64", "ubuntu-24.04-amd64", id="special-characters"
```

Comment:
```suggestion
            "ubuntu@24.04:amd64", "ubuntu-24.04-amd64", "𝔣𝔯𝔞𝔨𝔱𝔲𝔯'𝔡 𝔭𝔩𝔞𝔱𝔣𝔬𝔯𝔪", id="special-characters"
```

Reply:
I think this will probably fail, but we need to be able to do something that converts the platform name into something safe.

Reply:
Good catch.  In this scenario, craft-providers will raise the error:
```
failed to create an instance with name '𝔣𝔯𝔞𝔨𝔱𝔲𝔯'𝔡 𝔭𝔩𝔞𝔱𝔣𝔬𝔯𝔪'
name must contain at least one alphanumeric character
```

I added this test to craft-providers and found it a bug, so thank you!

https://github.com/canonical/craft-providers/pull/712/commits/ff32f58632692fadfd8523ca01478be945810532